### ISSUE TITLE
When fade:true afterChange returns wrong index

### ISSUE THREAD

**Issue by Yuripetusko** [akiran/react-slick#115]
Hello, it's a weird one. ONLY if I set fade: true my afterChange function returns wrong index. I get 0, 1, 1, 2, 2, 3, 3..

So looks like I get previous index sometimes, maybe it is something to do with absence of webkitTransitionComplete with fade?

```
const carouselSettingsBig = {
      dots: false,
      infinite: true,
      speed: 500,
      fade: true,
      slidesToShow: 1,
      slidesToScroll: 1,
      lazyLoad: false,
      initialSlide: this.state.currentSlide,
      afterChange: function(index) {
        console.log('currentSlide', index)
        this.setState({ currentSlide: index });
      }.bind(this)
    };
```

**Comment by Yuripetusko** [comment]
A I know what's the issue is. it's in  helpers.js line 131 should be _this.props.afterChange(targetSlide); will create PR now

**Comment by diessica** [comment]
Fixed in #358
